### PR TITLE
Attempting to fix building everything when VS 2019 is the only one installed

### PR DESCRIPTION
**Pull request by agr**
Extracted part of the logic of `Get-MsBuildExe` function into `Get-MsBuildExeInternal` so I could try a bunch of things and `return` as soon as we find something. Adjusted the fallback in the original function accordingly.

### CODE REVIEW

File: build/common.ps1 [hunk 1]
```diff
@@ -77,17 +77,51 @@ Function Clear-Artifacts {
...
+{
+    if ($MSBuildVersion -lt 15) {
+        $MSBuildExe = Join-Path $MSBuildRoot ([string]$MSBuildVersion + ".0")
+        $MSBuildPath = Join-Path $MSBuildExe $MSBuildExeRelPath
+        return $MSBuildPath;
+    } elseif ($MSBuildVersion -eq 16) {
+        $pathsToCheck = @(
+            "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe",
+            "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\MSBuild.exe",
+            "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe",
```
Comment (agr) [inline build/common.ps1]: Need to verify that top 3 paths are actually found in the wild.
Bottom 2 definitely are.

Comment (joelverhagen) [reply build/common.ps1]: At this point, do you think something link `vswhere` is a better approach?
https://github.com/microsoft/vswhere

Comment (agr) [reply build/common.ps1]: Sounds reasonable

Comment (agr) [reply build/common.ps1]: On the other hand, after talking to @scottbommarito, I made much simpler and more correct update, which should be OK.

File: build/common.ps1 [hunk 2]
```diff
@@ -107,9 +107,11 @@ Function Get-MSBuildExe {
         }
         
         $installations = @([NuGet.Services.Build.VisualStudioSetupConfigurationHelper]::GetInstancePaths() | ForEach-Object {
-            $MSBuildRoot = Join-Path "$_\MSBuild" ([string]$MSBuildVersion + ".0")
-            Join-Path $MSBuildRoot $MSBuildExeRelPath
-        } | Where-Object { Test-Path $_ })
+            Join-Path "$_\MSBuild" ([string]$MSBuildVersion + ".0")
```
Comment (loic-sharma) [inline build/common.ps1]: Use `[IO.Path]::Combine(...)` when there's more than two path fragments

**Pull request closed without merging**